src/node_type.rs

```diff
@@ -98,54 +117,21 @@ impl NodeKey {
     pub(crate) fn set_version(&mut self, version: Version) {
         self.version = version;
     }
-
     /// Serializes to bytes for physical storage enforcing the same order as that in memory.
+    ///
+    /// BorshSerialize::Seserialize instead
     pub fn encode(&self) -> Result<Vec<u8>> {
-        let mut out = vec![];
-        out.write_u64::<BigEndian>(self.version())?;
-        out.write_u8(self.nibble_path().num_nibbles() as u8)?;
-        out.write_all(self.nibble_path().bytes())?;
-        Ok(out)
+        self.try_to_vec().map_err(|e| e.into())
     }
-
     /// Recovers from serialized bytes in physical storage.
+    ///
+    /// Provided for backwards compatibility. Prefer using BorshDeserialize::deserialize instead
     pub fn decode(val: &[u8]) -> Result<NodeKey> {
```

Comment:
> A couple of thoughts about backwards compatibility: I think we do not need to preserve it, and we should break it while we can (i.e. while there are not downstream users other than ourselves). What do you think?

You're completely right, we might as well enjoy breaking the APIs as much as possible since there are no real consumers! This will break a few things in Sovereign, but it's an easy fix. I'll remove all of the backwards compatibility cruft :)